File: android/src/main/java/com/reactnativeescposprinter/ThePrinterManager.java
package com.escposprinter;

import android.util.ArrayMap;
import android.util.Log;
import com.escposprinter.ThePrinter;

public class ThePrinterManager {

    static final private ArrayMap<String, ThePrinter> objectDict_ = new ArrayMap<>();

    private static ThePrinterManager instance = null;

    private ThePrinterManager() {
        // Exists only to defeat instantiation.
    }

    synchronized public static ThePrinterManager getInstance() {
        if (instance == null) {
            instance = new ThePrinterManager();
        }

        return instance;
    }

    synchronized public String add(final ThePrinter obj, final String printerTarget) {
        if (!objectDict_.containsKey(printerTarget)) {
            objectDict_.put(printerTarget, obj);
        }

        return printerTarget;
    }

    synchronized public void remove(final String printerTarget) {
        objectDict_.remove(printerTarget);
    }

    synchronized public ThePrinter getObject(final String printerTarget) {
        return objectDict_.get(printerTarget);
    }

}

File: package/ios/ThePrinterManager.m
#import "ThePrinterManager.h"

@interface ThePrinterManager()
                                        // objectid  thePrinter
@property(nonatomic) NSMutableDictionary<NSString *, ThePrinter*> *objectDict;

@end


@implementation ThePrinterManager

static ThePrinterManager *sharedData_ = nil;

+ (ThePrinterManager *)sharedManager {
    @synchronized(self) {
        if (sharedData_ == nil) {
            sharedData_ = [ThePrinterManager new];
        }
        return sharedData_;
    }
}

- (instancetype)init {
    self = [super init];
    if (self) {
        _objectDict = [NSMutableDictionary new];
    }

    return self;
}

- (void) removeAll{
    @synchronized (self) {
        if (_objectDict == nil) return;
        [_objectDict removeAllObjects];
    }
}

- (NSString *)add:(ThePrinter*)thePrinterObj {
    @synchronized (self) {
        NSString *printerTarget = [thePrinterObj getPrinterTarget];

        for (NSString *aKey in [_objectDict allKeys] )
        {
            NSLog(@"Here is the key: %@", aKey);
        }
        if ([_objectDict.allKeys containsObject:printerTarget] == NO) {
            NSLog(@"Setting the instance for the printer");
            [_objectDict addEntriesFromDictionary:@{printerTarget: thePrinterObj}];
        } else {
            NSLog(@"The key already exists");
        }
        return printerTarget;
    }
}

- (ThePrinter*)getObject:(NSString *) identifier {
    @synchronized (self) {
        return _objectDict[identifier];
    }
}

- (void)remove:(NSString *) identifier {
    @synchronized (self) {
        _objectDict[identifier] = nil;
    }
}


@end
